tests/unit/synapseclient/core/download/unit_test_download_functions.py

```diff
@@ -0,0 +1,42 @@
+"""Unit tests for synapseclient.core.download.download_functions"""
```

Comment:
Might be a good idea to test out `resolve_download_path_collisions`

Reply:
I added integration tests around these different file collision types in: https://github.com/Sage-Bionetworks/synapsePythonClient/pull/1106/files#diff-e1692a03c6aa3f388426318480c8b7405e995d62eec5a19098876ebf3a5953c5